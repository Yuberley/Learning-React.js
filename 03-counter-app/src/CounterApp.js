import React, { useState } from 'react';
import PropTypes from 'prop-types';


const CounterApp = ({value}) => { 

    const [ counter, setCounter ] = useState( value );

    return (
        <>
            <h1>Counter App</h1>
            <h2> { counter } </h2>
            <button onClick={ () => setCounter( value )}>Reset</button>
            <button onClick={ () => setCounter( counter + 1 )}>+1</button>
            <button onClick={ () => setCounter( counter - 1 )}>-1</button>
        </>
    );

}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

const RelojApp = () => {
    const [hora,setHora]=useState(reloj);
    setTimeout(() => {
      setHora(reloj)
    }, 1000);
    return (
      <>
        <span>{hora}</span>
      </>
    )
  }
  const reloj = () => {
    let date = new Date();
    let horaActual = {
      hora: date.getHours(),
      minuto: date.getMinutes(),
      segundo: date.getSeconds()
    }
    const { hora, minuto, segundo } = horaActual;
    const now = `${hora}:${minuto < 10 ? '0' + minuto : minuto}:${segundo<10? '0' + segundo : segundo}`;
    return now;
  }
   
export {
    CounterApp as default,
    RelojApp,
}