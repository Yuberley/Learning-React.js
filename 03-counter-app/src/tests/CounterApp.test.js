import React from 'react';
import '@testing-library/jest-dom';
import CounterApp from '../CounterApp';
import { shallow } from 'enzyme';

describe('Test on file CounterApp', () => {

    let wrapper;

    beforeEach( () => {

        wrapper = shallow(<CounterApp value = {0} />);

    })

    test('should show the component successfully', () => {
        
        expect( wrapper ).toMatchSnapshot();

    })

    test('should show default value 100', () => {

        const wrapper = shallow(<CounterApp value = {100} />);
        const valueDefault = wrapper.find('h2').text().trim();

        expect( valueDefault ).toBe( '100' );

    })

    test('should increment counter +1', () => {
        
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe( '1' );

    })

    test('should decrement counter -1', () => {

        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe( '-1' );

    })

    test('should reset counter 0', () => {

        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe( '0' );

    })


})