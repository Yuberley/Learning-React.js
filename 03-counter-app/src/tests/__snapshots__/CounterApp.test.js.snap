// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on file CounterApp should show the component successfully 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
     
    0
     
  </h2>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
